test/test_benchmark.py

```diff
@@ -903,6 +931,36 @@ def output_parser(output):
       return float(re.search(r'Total elapsed: ([\d\.]+)', output).group(1))
     self.do_benchmark('matrix_multiply', read_file(test_file('matrix_multiply.cpp')), 'Total elapsed:', output_parser=output_parser, shared_args=['-I' + test_file('benchmark')])
 
+  @non_core
+  def test_sizes(self):
+    if len(benchmarkers) != 1 or not isinstance(benchmarkers[0], EmscriptenBenchmarker):
+      self.skipTest('size benchmark only runs on emscripten')
+    benchmarkers[0].skip_bench = True
+    output = {
+      'version': 1,
+      'git_hash': '',
+      'results': []
+    }
+
+    def run_sub_test(name, testFn):
+      result = {
+        'key': {
+          'test': name,
+          'units': 'bytes'
+        },
+        'measurements': {
+          'stats': []
+        }
+      }
+      # The benchmark will add to this attribute.
+      benchmarkers[0].size_measurements = result['measurements']['stats']
+      testFn()
+      output['results'].append(result)
+
+    run_sub_test('primes', self.test_primes)
+    run_sub_test('primes_nocheck', self.test_primes_nocheck)
```

Comment:
This running of a test method from another test method seems odd to me.

Is there not perhaps some way we could use a decorator which we add to `primes` and `primes_nocheck`?   Something like `@track_size`?  Then the track_size decorator could take care of doing all of this?   

Then when the suite is run in "track sizes" mode we could skip any tests that don't have the decorator?

Reply:
From emscripten-releases I want to run one command that will run all the sizes tests so we don't have to update emscripten-releases each time we add another size benchmark. Do you have any thoughts on how to achieve that with a decorator? I suppose I could add a parameterized version of each test, so then I could run `test/runner benchmark.test_*_sizes` or something.

Reply:
I was thinking that you would just run the entire branchmark suite.

In sizes mode we would just skip any benchmarks that are not annotated.. and then adding new tests would be a simple as adding the annotation.

Reply:
I think I've come up with something better. Now you can use the 'size' benchmarker and it will record stats for any test you run and save them on tear down. This will require us to specify which benchmarks we're recording in emscripten-releases, but I think that's fine since it gets rid of the weird meta test here and I don't imagine the list would change much.